### DISCUSSION TITLE
Basic example of filtering through results

### DISCUSSION THREAD

**Discussion by bogdan-ciocoiu** [FriendsOfTYPO3/extension_builder#668] [Q&A] [answered]
I am beginning my journey with MVC frameworks - I apologise for the newbie question. I have developed an extension using Extension Builder that shows (for example) a collection of Books and one of the Authors. There is an apparent one-to-many correlation between those.

I am looking for basic examples readily available on the Internet of similar objects built with Extension Builder that I can analyse to develop a search functionality. The search functionality I envisage is a keyword-based search that consumes either the 'summary' property of a book or the 'name' property of an author.

Thank you.

**Comment by DavidBruchmann** [comment, marked as answer]
I just wanna point out that while it's perfectly fine to use this scenario for training, there exist books with several authors.  
You might want to consider this for the database correlation.

This is an example extension that is mentioned also in some documentation: https://github.com/TYPO3-Documentation/blog_example

Furthermore you could install the styleguide extension to get an overview of many configuration options concerning fields in the backend forms, which also concerns the fields in the database tables: https://github.com/TYPO3-CMS/styleguide

**Reply by bogdan-ciocoiu** [reply]
That's exactly what I needed, thank you. Also, thank you for the testing extension.

**Reply by bogdan-ciocoiu** [reply]
The blog example is very complex for me. It will take a fair amount of time until I get to the bottom of it.

I cannot pass a $_POST variable from a viewer to a method in the desired controller. Most online resources suggest that just defining the variable as exemplified below will work (provided the name of the $variable is the same as the name of the input in the form.

` public function processFormAction($myValue = '')`

 I also tried using `getArguments` as below

` $myValue = $this->request->getArguments()['myValue'] ?? '';`

This is how I defined the form:

` <form action="{f:uri.action(action: 'search', controller: 'Books')}" method="post">`

I am trying to avoid manually handling the $_POST["myValue"] variable in the method; it is not the best practice, and I would benefit from Typo3's sanitisation instead. Plus, I want to learn how to do this properly. 

Is there any additional mapping or anything one must do to get the method to recognise the post variable?

**Reply by bogdan-ciocoiu** [reply]
Problem solved.

**Viewer:**

```
    <f:form action='filter' object="Services" objectName="services">
        <f:form.submit value="Search" class="btn btn-primary"  />
        <f:form.textfield property="keyword" class="form-control" id="keyword" />
    </f:form>
```

**Controller:**

```
    public function filterAction(): \Psr\Http\Message\ResponseInterface
    {
        $keyword = $this->request->getArguments()["services"]["keyword"];
        // More logic goes here.
        return $this->htmlResponse();
    }
```
